app/main.py

```diff
@@ -1,6 +1,24 @@
 from typing import Callable
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    cache_dict = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs) -> Callable:
+
+        key = (args, tuple(sorted(kwargs.items())))
+
+        if key in cache_dict:
+            print("Getting from cache")
+            return cache_dict[key]
+        else:
+            print("Calculating new result")
+            result = func(*args, **kwargs)
+            cache_dict[key] = result
+            return result
+
+    wrapper.cache = cache_dict
```

Comment:
Attaching the `cache_dict` to the `wrapper` function as an attribute is a clever way to expose the cache, but it might be better to use a more descriptive name like `wrapper._cache` to indicate that it's intended for internal use.